Reply to comment (merge comments + answers) feature

ViditChitkara opened the pull request:
Fixes #4094

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] PR is descriptively titled 📑
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

SidharthBansal commented:
@ViditChitkara there are some conflicts, can you please rebase your changes?

ViditChitkara commented:
@SidharthBansal sure, will do!!

ViditChitkara commented:
@jywarren , the system is working fine but with each comment a "reply to this" comment box appears and according to me it is making our 'note' or 'question' page quite big. So, shall we hide the 'reply to comment' form in the beginning and just give a toggle link to open the reply form?

jywarren commented:
Yeah, I think we can have just a toggle button to open the response form.
Can you screenshot once you change it?

Have you looked at how node.answers is counted because we track how many
responses there are to each question... Will that now just count comments?

Thanks!!!! Awesome!!!



On Sun, Mar 3, 2019, 5:01 PM Vidit <notifications@github.com> wrote:

> @jywarren <https://github.com/jywarren> , the system is working fine but
> with each comment a "reply to this" comment box appears and according to me
> it is making our 'note' or 'question' page quite big. So, shall we hide the
> 'reply to comment' form in the beginning and just give a toggle link to
> open the reply form?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469069234>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ7eROvaxV8cLnlw8EiYtEnZ3Cl6eks5vTEYUgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Sure, will do once I change it. Answer count is indicated by
@node.answers.length now that answers will be deprecated, only comments
will count. Also the comment count will be a total of normal comments and
replied comments as well.
How does this sound??

Thanks

On Mon, Mar 4, 2019, 4:50 AM Jeffrey Warren <notifications@github.com>
wrote:

> Yeah, I think we can have just a toggle button to open the response form.
> Can you screenshot once you change it?
>
> Have you looked at how node.answers is counted because we track how many
> responses there are to each question... Will that now just count comments?
>
> Thanks!!!! Awesome!!!
>
>
>
> On Sun, Mar 3, 2019, 5:01 PM Vidit <notifications@github.com> wrote:
>
> > @jywarren <https://github.com/jywarren> , the system is working fine but
> > with each comment a "reply to this" comment box appears and according to
> me
> > it is making our 'note' or 'question' page quite big. So, shall we hide
> the
> > 'reply to comment' form in the beginning and just give a toggle link to
> > open the reply form?
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-469069234>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJ7eROvaxV8cLnlw8EiYtEnZ3Cl6eks5vTEYUgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469076074>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVkX0YycxWNCfpiuzXfArM5Ii7BHi0xQks5vTFikgaJpZM4ZyBu4>
> .
>

jywarren commented:
Sounds great!

On Mon, Mar 4, 2019, 1:53 AM Vidit <notifications@github.com> wrote:

> Sure, will do once I change it. Answer count is indicated by
> @node.answers.length now that answers will be deprecated, only comments
> will count. Also the comment count will be a total of normal comments and
> replied comments as well.
> How does this sound??
>
> Thanks
>
> On Mon, Mar 4, 2019, 4:50 AM Jeffrey Warren <notifications@github.com>
> wrote:
>
> > Yeah, I think we can have just a toggle button to open the response form.
> > Can you screenshot once you change it?
> >
> > Have you looked at how node.answers is counted because we track how many
> > responses there are to each question... Will that now just count
> comments?
> >
> > Thanks!!!! Awesome!!!
> >
> >
> >
> > On Sun, Mar 3, 2019, 5:01 PM Vidit <notifications@github.com> wrote:
> >
> > > @jywarren <https://github.com/jywarren> , the system is working fine
> but
> > > with each comment a "reply to this" comment box appears and according
> to
> > me
> > > it is making our 'note' or 'question' page quite big. So, shall we hide
> > the
> > > 'reply to comment' form in the beginning and just give a toggle link to
> > > open the reply form?
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/publiclab/plots2/pull/4537#issuecomment-469069234
> >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AABfJ7eROvaxV8cLnlw8EiYtEnZ3Cl6eks5vTEYUgaJpZM4ZyBu4
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-469076074>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AVkX0YycxWNCfpiuzXfArM5Ii7BHi0xQks5vTFikgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469138862>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ6Oh51e40HV0i42BLY4-CjAOmIkCks5vTMK4gaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
<img width="1440" alt="screenshot 2019-03-05 at 2 22 29 am" src="https://user-images.githubusercontent.com/22616017/53762530-5de5f180-3eee-11e9-9e13-2d72dfe7666f.png">
<img width="1440" alt="screenshot 2019-03-05 at 2 31 47 am" src="https://user-images.githubusercontent.com/22616017/53762731-de0c5700-3eee-11e9-8b32-c6454be36d14.png">
@jywarren here is how the toggle text will look like..this also shows how replied comments will be shown...what do you think?

PS: This is some test data on my local machine

jywarren commented:
this looks great!

On Mon, Mar 4, 2019 at 4:08 PM Vidit <notifications@github.com> wrote:

> [image: screenshot 2019-03-05 at 2 22 29 am]
> <https://user-images.githubusercontent.com/22616017/53762530-5de5f180-3eee-11e9-9e13-2d72dfe7666f.png>
>
> [image: screenshot 2019-03-05 at 2 31 47 am]
> <https://user-images.githubusercontent.com/22616017/53762731-de0c5700-3eee-11e9-8b32-c6454be36d14.png>
> @jywarren <https://github.com/jywarren> here is how the toggle text will
> look like..this also shows how replied comments will be shown...what do you
> think?
>
> PS: This is some test data on my local machine
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469422231>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ5PxAGOwfiOZU-56M0O8ULCHMnJMks5vTYtDgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Cool, I'll be making a few changes and then I guess it will be good to go!!

On Tue, Mar 5, 2019, 3:03 AM Jeffrey Warren <notifications@github.com>
wrote:

> this looks great!
>
> On Mon, Mar 4, 2019 at 4:08 PM Vidit <notifications@github.com> wrote:
>
> > [image: screenshot 2019-03-05 at 2 22 29 am]
> > <
> https://user-images.githubusercontent.com/22616017/53762530-5de5f180-3eee-11e9-9e13-2d72dfe7666f.png
> >
> >
> > [image: screenshot 2019-03-05 at 2 31 47 am]
> > <
> https://user-images.githubusercontent.com/22616017/53762731-de0c5700-3eee-11e9-8b32-c6454be36d14.png
> >
> > @jywarren <https://github.com/jywarren> here is how the toggle text will
> > look like..this also shows how replied comments will be shown...what do
> you
> > think?
> >
> > PS: This is some test data on my local machine
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-469422231>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJ5PxAGOwfiOZU-56M0O8ULCHMnJMks5vTYtDgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469430443>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVkX0WNc-64mbx5nOw8hmH9m0O-VRFTMks5vTZFGgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
@jywarren , I think a basic part of the issue is completed. Would you mind reviewing this PR and suggest any changes. Just wanted to know if I'm going in the right direction.

ViditChitkara commented:
The parts which I think are completed are:-
1. Migration to convert answers to questions (I have still not deleted the answers which will be migrated. Maybe we can use them if needed in future).
2. Reply to comments feature. Added a reply_to field to comments and made 'aid' for all comments to be 0. Later on the aid column will be removed from comments.
3. Removed just the answer form and answers section from question/show page. This means that users will only be able to comment on a question.
@jywarren how does this sound?

jywarren commented:
This is looking good - what parts do you think still need work?

On Tue, Mar 5, 2019 at 11:50 PM Vidit <notifications@github.com> wrote:

> The parts which I think are completed are:-
>
>    1. Migration to convert answers to questions (I have still not deleted
>    the answers which will be migrated. Maybe we can use them if needed in
>    future).
>    2. Reply to comments feature. Added a reply_to field to comments and
>    made 'aid' for all comments to be 0. Later on the aid column will be
>    removed from comments.
>    3. Removed just the answer form and answers section from question/show
>    page. This means that users will only be able to comment on a question.
>    @jywarren <https://github.com/jywarren> how does this sound?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-469967359>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ2bp1WvNDVtfsA1d-QT0OcKbAvnIks5vT0jbgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
I think we need to work on the following parts:-

1. migrate answer_selection (i.e. likes on answers) to the differently-plumbed comment likes (which is actually a nicer system with emoji and such). For this, will a like via "ThumbsUp" emoji do?

2. figure out how the question.answers features will work, because answers can be "accepted" -- perhaps this also becomes part of the comment liking system, with a "checkbox" emoji? 
We can surely add a "checkbox" emoji to comments which were earlier accepted answers, but since we will not be having answers in future, will this emoji make sense on comments?

3. ensure statistics for Answers (http://publiclab.org/stats) are preserved somehow or merged into comments?

@jywarren what do you think?
A lot of tests are failing due to usage of "reply_to" column in comments, working to fix that.

jywarren commented:
OK. So, under the current state, we don't yet have to worry about stats for
answers because we aren't yet deleting any. Same with question.answers.

And people will still be able to post answers under the current state, or
have we turned that off?

And so the remaining issue would be to actually convert answers previously
posted into comments, and as part of that, make sure the answers stats and
questions.answers are both properly preserved?

Thank you, this is helpful!

On Wed, Mar 6, 2019 at 4:10 PM Vidit <notifications@github.com> wrote:

> I think we need to work on the following parts:-
>
>    1.
>
>    migrate answer_selection (i.e. likes on answers) to the
>    differently-plumbed comment likes (which is actually a nicer system with
>    emoji and such). For this, will a like via "ThumbsUp" emoji do?
>    2.
>
>    figure out how the question.answers features will work, because
>    answers can be "accepted" -- perhaps this also becomes part of the comment
>    liking system, with a "checkbox" emoji?
>    We can surely add a "checkbox" emoji to comments which were earlier
>    accepted answers, but since we will not be having answers in future, will
>    this emoji make sense on comments?
>    3.
>
>    ensure statistics for Answers (http://publiclab.org/stats) are
>    preserved somehow or merged into comments?
>
> @jywarren <https://github.com/jywarren> what do you think?
> A lot of tests are failing due to usage of "reply_to" column in comments,
> working to fix that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-470277692>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJwyBpVvssXbt7C_T-vMBxRhqTs7kks5vUC6agaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
@jywarren , we have turned off answer postings. Otherwise there will again be some answers which will have to be converted to comments. Other issues are complete with this pr I guess, so shall we test it somewhere?

jywarren commented:
Yes, awesome, we can test on unstable - do you know how and do you need permissions to push to the unstable branch? 

Are we copying answer likes to be comment likes yet? I wouldn't want to lose existing information about those. 

Finally, I agree we should leave the answers table in place until we are 100% sure we have successfully copied all the information into the comments system with no losses. Then we can get rid of the table and model in follow-up!

Why don't you test it on unstable and see -- the other thing you can do is do `puts "Answers: " + Answer.count.to_s`  and "Comments: " + Comment.count.to_s` before and after the migration to have a record in the log of how many were converted successfully. How does that sound?

ViditChitkara commented:
@jywarren , sure I'll test in on unstable. Yes, we are creating Comment Likes ("ThumbsUp emoji") for answer likes. However, nothing is deleted from the db. Even the comments which belong to answers will be available, since I'm just changing the reply_to field value in them, if anything wrong occurs we can easily get back to our older system.

jywarren commented:
super!

On Thu, Mar 7, 2019 at 2:29 PM Vidit <notifications@github.com> wrote:

> @jywarren <https://github.com/jywarren> , sure I'll test in on unstable.
> Yes, we are creating Comment Likes ("ThumbsUp emoji") for answer likes.
> However, nothing is deleted from the db. Even the comments which belong to
> answers will be available, since I'm just changing the reply_to field value
> in them, if anything wrong occurs we can easily get back to our older
> system.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-470661589>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ87YgxTVj9hgtPy0ciBlMrSyWVPVks5vUWiRgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
I pushed this on unstable. It's working on unstable.publiclab.org . Kindly
have a look! I think it's ready to be merged now.

Thanks!

On Fri, Mar 8, 2019, 1:06 AM Jeffrey Warren <notifications@github.com>
wrote:

> super!
>
> On Thu, Mar 7, 2019 at 2:29 PM Vidit <notifications@github.com> wrote:
>
> > @jywarren <https://github.com/jywarren> , sure I'll test in on unstable.
> > Yes, we are creating Comment Likes ("ThumbsUp emoji") for answer likes.
> > However, nothing is deleted from the db. Even the comments which belong
> to
> > answers will be available, since I'm just changing the reply_to field
> value
> > in them, if anything wrong occurs we can easily get back to our older
> > system.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-470661589>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJ87YgxTVj9hgtPy0ciBlMrSyWVPVks5vUWiRgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-470664175>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVkX0dXXpwkCnft0YI6UkYB14OCn_d4jks5vUWpRgaJpZM4ZyBu4>
> .
>

jywarren commented:
Oh but it is failing for some reason? Can you take another look before we move forward? Thanks!!!

ViditChitkara commented:
Sure, will have a look!

ViditChitkara commented:
@jywarren the tests are passing now!! Have a look! Also added tests for replied comments.

jywarren commented on public/cron_log.log:
```diff
@@ -0,0 +1,9 @@
+Thu Mar 14 18:41:00 UTC 2019
```
Oh oops, maybe we don't want to include this file?

ViditChitkara replied:
Will do!

jywarren commented:
This is super! Sorry just one more file to remove, then this is ready to merge! 🎉

ViditChitkara commented:
@jywarren done!

jywarren commented:
Great! Let's check carefully on stable.publiclab.org once it builds there.

ViditChitkara commented:
Have a look at the comments here. Some of them may have been answers which are converted to comments and corresponding comments are converted into replied comments!
https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
This is really exciting!

ViditChitkara commented:
Should I start with 'removing answer related code' now as a follow-up of this?

jywarren commented:
Sure, but wait, it seems like some answers don't exist on stable --
https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849

was there some kind of error?

On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com> wrote:

> Have a look at the comments here. Some of them may have been answers which
> are converted to comments and corresponding comments are converted into
> replied comments!
>
> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
> This is really exciting!
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
This is a bit odd!! At least the answer section should have been
removed.Also, I don't see the updated code over here. At least the comments
should have 'emoji like' feature which I think is not there. Will have a
look at it. Is there some way we could have a look at how the migrations
ran? Any sort of logs?
Thanks!!

On Tue, Mar 19, 2019, 7:54 PM Jeffrey Warren <notifications@github.com>
wrote:

> Sure, but wait, it seems like some answers don't exist on stable --
>
> https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
>
> was there some kind of error?
>
> On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com> wrote:
>
> > Have a look at the comments here. Some of them may have been answers
> which
> > are converted to comments and corresponding comments are converted into
> > replied comments!
> >
> >
> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
> > This is really exciting!
> >
> > —
> > You are receiving this because you modified the open/close state.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474393332>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVkX0bSjZ7JEe1vGo8Jw-F8UNZWkSejkks5vYPMMgaJpZM4ZyBu4>
> .
>

jywarren commented:
Yes, i think here?
https://jenkins.laboratoriopublico.org/job/Plots-Stable/lastBuild/console

On Tue, Mar 19, 2019 at 10:36 AM Vidit <notifications@github.com> wrote:

> This is a bit odd!! At least the answer section should have been
> removed.Also, I don't see the updated code over here. At least the comments
> should have 'emoji like' feature which I think is not there. Will have a
> look at it. Is there some way we could have a look at how the migrations
> ran? Any sort of logs?
> Thanks!!
>
> On Tue, Mar 19, 2019, 7:54 PM Jeffrey Warren <notifications@github.com>
> wrote:
>
> > Sure, but wait, it seems like some answers don't exist on stable --
> >
> >
> https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
> >
> > was there some kind of error?
> >
> > On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com> wrote:
> >
> > > Have a look at the comments here. Some of them may have been answers
> > which
> > > are converted to comments and corresponding comments are converted into
> > > replied comments!
> > >
> > >
> >
> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
> > > This is really exciting!
> > >
> > > —
> > > You are receiving this because you modified the open/close state.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306
> >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474393332>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AVkX0bSjZ7JEe1vGo8Jw-F8UNZWkSejkks5vYPMMgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474398587>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ85H6sJZi-z7uy6vhZoFq5ke3jRPks5vYPXOgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Will have a look!

On Tue, Mar 19, 2019, 8:14 PM Jeffrey Warren <notifications@github.com>
wrote:

> Yes, i think here?
> https://jenkins.laboratoriopublico.org/job/Plots-Stable/lastBuild/console
>
> On Tue, Mar 19, 2019 at 10:36 AM Vidit <notifications@github.com> wrote:
>
> > This is a bit odd!! At least the answer section should have been
> > removed.Also, I don't see the updated code over here. At least the
> comments
> > should have 'emoji like' feature which I think is not there. Will have a
> > look at it. Is there some way we could have a look at how the migrations
> > ran? Any sort of logs?
> > Thanks!!
> >
> > On Tue, Mar 19, 2019, 7:54 PM Jeffrey Warren <notifications@github.com>
> > wrote:
> >
> > > Sure, but wait, it seems like some answers don't exist on stable --
> > >
> > >
> >
> https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
> > >
> > > was there some kind of error?
> > >
> > > On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com>
> wrote:
> > >
> > > > Have a look at the comments here. Some of them may have been answers
> > > which
> > > > are converted to comments and corresponding comments are converted
> into
> > > > replied comments!
> > > >
> > > >
> > >
> >
> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
> > > > This is really exciting!
> > > >
> > > > —
> > > > You are receiving this because you modified the open/close state.
> > > > Reply to this email directly, view it on GitHub
> > > > <
> https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306
> > >,
> > > > or mute the thread
> > > > <
> > >
> >
> https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4
> > > >
> > > > .
> > > >
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474393332
> >,
> > > or mute the thread
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/AVkX0bSjZ7JEe1vGo8Jw-F8UNZWkSejkks5vYPMMgaJpZM4ZyBu4
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474398587>,
> > or mute the thread
> > <
> https://github.com/notifications/unsubscribe-auth/AABfJ85H6sJZi-z7uy6vhZoFq5ke3jRPks5vYPXOgaJpZM4ZyBu4
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474402559>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AVkX0Vi21cqNlGw-XB-58JbZjl2yP8_Rks5vYPfagaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
@Jeffrey Warren <jywarren@gmail.com> are we sure that the data on stable
was synced with production? Bcoz I do not see a 5 month old comment (which
is on production) over stable.
And the pattern suggests that last 3 answers (which are 1 year old are not
there). However, all older answers are converted to comments. Is there some
means by which we could sync the latest production data with stable and
then test?

On Tue, Mar 19, 2019 at 9:56 PM Vidit Chitkara <chitkara.vidit@gmail.com>
wrote:

> Will have a look!
>
> On Tue, Mar 19, 2019, 8:14 PM Jeffrey Warren <notifications@github.com>
> wrote:
>
>> Yes, i think here?
>> https://jenkins.laboratoriopublico.org/job/Plots-Stable/lastBuild/console
>>
>> On Tue, Mar 19, 2019 at 10:36 AM Vidit <notifications@github.com> wrote:
>>
>> > This is a bit odd!! At least the answer section should have been
>> > removed.Also, I don't see the updated code over here. At least the
>> comments
>> > should have 'emoji like' feature which I think is not there. Will have a
>> > look at it. Is there some way we could have a look at how the migrations
>> > ran? Any sort of logs?
>> > Thanks!!
>> >
>> > On Tue, Mar 19, 2019, 7:54 PM Jeffrey Warren <notifications@github.com>
>> > wrote:
>> >
>> > > Sure, but wait, it seems like some answers don't exist on stable --
>> > >
>> > >
>> >
>> https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
>> > >
>> > > was there some kind of error?
>> > >
>> > > On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com>
>> wrote:
>> > >
>> > > > Have a look at the comments here. Some of them may have been answers
>> > > which
>> > > > are converted to comments and corresponding comments are converted
>> into
>> > > > replied comments!
>> > > >
>> > > >
>> > >
>> >
>> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
>> > > > This is really exciting!
>> > > >
>> > > > —
>> > > > You are receiving this because you modified the open/close state.
>> > > > Reply to this email directly, view it on GitHub
>> > > > <
>> https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306
>> > >,
>> > > > or mute the thread
>> > > > <
>> > >
>> >
>> https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4
>> > > >
>> > > > .
>> > > >
>> > >
>> > > —
>> > > You are receiving this because you were mentioned.
>> > > Reply to this email directly, view it on GitHub
>> > > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474393332
>> >,
>> > > or mute the thread
>> > > <
>> >
>> https://github.com/notifications/unsubscribe-auth/AVkX0bSjZ7JEe1vGo8Jw-F8UNZWkSejkks5vYPMMgaJpZM4ZyBu4
>> > >
>> > > .
>> > >
>> >
>> > —
>> > You are receiving this because you were mentioned.
>> > Reply to this email directly, view it on GitHub
>> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474398587>,
>> > or mute the thread
>> > <
>> https://github.com/notifications/unsubscribe-auth/AABfJ85H6sJZi-z7uy6vhZoFq5ke3jRPks5vYPXOgaJpZM4ZyBu4
>> >
>> > .
>> >
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474402559>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AVkX0Vi21cqNlGw-XB-58JbZjl2yP8_Rks5vYPfagaJpZM4ZyBu4>
>> .
>>
>

jywarren commented:
It's not fully synced, that's right! We could ask @icarito to do so, can
you contact him in the chatroom? Or, you could look on unstable, because
there should be the same "clone" of the database as stable, you know?

On Tue, Mar 19, 2019 at 3:02 PM Vidit <notifications@github.com> wrote:

> @Jeffrey Warren <jywarren@gmail.com> are we sure that the data on stable
> was synced with production? Bcoz I do not see a 5 month old comment (which
> is on production) over stable.
> And the pattern suggests that last 3 answers (which are 1 year old are not
> there). However, all older answers are converted to comments. Is there some
> means by which we could sync the latest production data with stable and
> then test?
>
> On Tue, Mar 19, 2019 at 9:56 PM Vidit Chitkara <chitkara.vidit@gmail.com>
> wrote:
>
> > Will have a look!
> >
> > On Tue, Mar 19, 2019, 8:14 PM Jeffrey Warren <notifications@github.com>
> > wrote:
> >
> >> Yes, i think here?
> >>
> https://jenkins.laboratoriopublico.org/job/Plots-Stable/lastBuild/console
> >>
> >> On Tue, Mar 19, 2019 at 10:36 AM Vidit <notifications@github.com>
> wrote:
> >>
> >> > This is a bit odd!! At least the answer section should have been
> >> > removed.Also, I don't see the updated code over here. At least the
> >> comments
> >> > should have 'emoji like' feature which I think is not there. Will
> have a
> >> > look at it. Is there some way we could have a look at how the
> migrations
> >> > ran? Any sort of logs?
> >> > Thanks!!
> >> >
> >> > On Tue, Mar 19, 2019, 7:54 PM Jeffrey Warren <
> notifications@github.com>
> >> > wrote:
> >> >
> >> > > Sure, but wait, it seems like some answers don't exist on stable --
> >> > >
> >> > >
> >> >
> >>
> https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
> >> > >
> >> > > was there some kind of error?
> >> > >
> >> > > On Tue, Mar 19, 2019 at 10:18 AM Vidit <notifications@github.com>
> >> wrote:
> >> > >
> >> > > > Have a look at the comments here. Some of them may have been
> answers
> >> > > which
> >> > > > are converted to comments and corresponding comments are converted
> >> into
> >> > > > replied comments!
> >> > > >
> >> > > >
> >> > >
> >> >
> >>
> https://stable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping
> >> > > > This is really exciting!
> >> > > >
> >> > > > —
> >> > > > You are receiving this because you modified the open/close state.
> >> > > > Reply to this email directly, view it on GitHub
> >> > > > <
> >> https://github.com/publiclab/plots2/pull/4537#issuecomment-474390306
> >> > >,
> >> > > > or mute the thread
> >> > > > <
> >> > >
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AABfJ2rbjSkwVs_CDzMPGnyDxx4XN_qEks5vYPF3gaJpZM4ZyBu4
> >> > > >
> >> > > > .
> >> > > >
> >> > >
> >> > > —
> >> > > You are receiving this because you were mentioned.
> >> > > Reply to this email directly, view it on GitHub
> >> > > <
> https://github.com/publiclab/plots2/pull/4537#issuecomment-474393332
> >> >,
> >> > > or mute the thread
> >> > > <
> >> >
> >>
> https://github.com/notifications/unsubscribe-auth/AVkX0bSjZ7JEe1vGo8Jw-F8UNZWkSejkks5vYPMMgaJpZM4ZyBu4
> >> > >
> >> > > .
> >> > >
> >> >
> >> > —
> >> > You are receiving this because you were mentioned.
> >> > Reply to this email directly, view it on GitHub
> >> > <https://github.com/publiclab/plots2/pull/4537#issuecomment-474398587
> >,
> >> > or mute the thread
> >> > <
> >>
> https://github.com/notifications/unsubscribe-auth/AABfJ85H6sJZi-z7uy6vhZoFq5ke3jRPks5vYPXOgaJpZM4ZyBu4
> >> >
> >> > .
> >> >
> >>
> >> —
> >> You are receiving this because you were mentioned.
> >> Reply to this email directly, view it on GitHub
> >> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474402559>,
> >> or mute the thread
> >> <
> https://github.com/notifications/unsubscribe-auth/AVkX0Vi21cqNlGw-XB-58JbZjl2yP8_Rks5vYPfagaJpZM4ZyBu4
> >
> >> .
> >>
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474527314>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ7mmQlGAhvL22LfLx1I0CFGrBncLks5vYTQwgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Actually I pushed this branch few days back to test on unstable so answers there would have been converted to comments. I guess we have got exactly the no. of comments here we want after conversion.
https://unstable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
However we can test it again on unstable. I'll ask @icarito about this!

jywarren commented:
you could also look at https://stable.publiclab.org/comments to see what
the most recent comments are that were copied!

On Tue, Mar 19, 2019 at 3:12 PM Vidit <notifications@github.com> wrote:

> Actually I pushed this branch few days back to test on unstable so answers
> there would have been converted to comments. I guess we have got exactly
> the no. of comments here we want after conversion.
>
> https://unstable.publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849
> However we can test it again on unstable. I'll ask @icarito
> <https://github.com/icarito> about this!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474532671>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJxCPwrnfO9nW1uZZILKHlXlS3qPJks5vYTaIgaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Will have a look!

ViditChitkara commented:
I guess this is showing comments in descending order of creation timestamp. Since we have preserved answer's timestamp, we can't really figure out which one's were created today! Maybe we'll need to re-sync stable's data and then run the migration in this branch. @icarito any ideas?

>you could also look at https://stable.publiclab.org/comments to see what
the most recent comments are that were copied!

jywarren commented:
OK, i think we are OK for this to go to production. I looked in the DB for the exact timestamps and this was one of the last comments in stable.publiclab.org before it was cloned: 

https://stable.publiclab.org/notes/gilbert/09-11-2017/hurricane-preparedness-dinner-discussion-sept-18-2017-6-pm-8-pm-cdt#c17361 -- it was on 12-09-2017, or Sept 12 2017. 

The missing comment by daisywatson here is from March 2018: https://publiclab.org/questions/warren/05-20-2017/cheap-lightweight-gopro-alternatives-for-aerial-mapping#a849

So it is not appearing because the db was copied before it was made. 

OK great!

ViditChitkara commented:
Great!

jywarren commented:
it's publishing now!

On Tue, Mar 19, 2019 at 10:10 PM Vidit <notifications@github.com> wrote:

> Great!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/publiclab/plots2/pull/4537#issuecomment-474655763>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABfJ8F9Uqduu8DKnV5ES7AO9Hi006Dxks5vYZh4gaJpZM4ZyBu4>
> .
>

ViditChitkara commented:
Fingers crossed!🤞
